fix(@embark/logger): Make `dir` the same as all other logger functions

emizzle opened the pull request:
### What did you refactor, implement, or fix?

During the latest round of Logger updates, the `writeToFile` function was changed to an `async.cargo` queue and all `writeToFile` references should have been updated to push a a task to the queue.

The main issue was that the `dir` function was not upgraded to push a task to the `writeToFile` queue.

#### How did you do it?

Upgrade the `dir` function to push a task to the `writeToFile` queue to be in line with all other logger functions.

#### Is there anything that needs special attention (breaking changes, etc)?


### Questions


### Review


### Cool Spaceship Picture

![Ancient UFO](https://theearthexpanded.files.wordpress.com/2014/11/mayan_spaceship1.jpg)

PascalPrecht commented on packages/core/logger/src/index.js:
```diff
@@ -197,13 +197,13 @@ export class Logger {
-    this.events.emit("log", "dir", obj);
-    this.logFunction(obj, null);
+    this.events.emit("log", "dir", args);
+    this.logFunction(args, null);
```
I might be wrong here, but I'm not sure we want `...args` here. `dir()` should take an object/map.

michaelsbradleyjr replied:
It's now correct to pass `args`.

See: https://github.com/embarklabs/embark/blob/fix/logger-dir/packages/core/logger/src/index.js#L26.

That was a refactor in my commit to PR #2184.

jrainville commented on packages/core/logger/src/index.js:
```diff
@@ -213,7 +213,7 @@ export class Logger {
       } catch (e) {}
     }
-    this.writeToFile({ args: obj, origin, prefix: "[dir]" });
+    this.writeToFile.push({ args, origin, prefix: "[dir]" });
```
Now that I think of it, should we print the `logger.dir` calls in the file? Are they gonna be useful to us? 
Asking, because in the releases, we don,t use `dir`, it's just for our own tests, so having those in the file might just be heavy for no reason.

michaelsbradleyjr commented on packages/core/logger/src/index.js:
```diff
@@ -197,13 +197,13 @@ export class Logger {
-  dir(obj) {
-    if (!obj || !(this.shouldLog('info'))) {
+  dir(...args) {
+    if (!args || !(this.shouldLog('info'))) {
```
If we're going with `...args` in the signature of `logger.dir` then the test should check `!args.length` instead of `!args` because even in the case of no arguments `args` will be `[]`, which is a truthy value.

But seem my [comment/question re: the change in signature]().

```suggestion
    if (!args.length || !(this.shouldLog('info'))) {
```

emizzle commented:
Reverted changes to the `dir` function, and removed the `writeToFile` call from `dir` altogether, as we don't need `dir` output written to the log file, only printed to the screen.
@PascalPrecht @michaelsbradleyjr @jrainville

PascalPrecht commented:
Nice! Merging.
